lib/concepts/school/operations/import_batch.rb

```diff
@@ -0,0 +1,163 @@
+        if duplicate_check[:error]
+          response[:error] = duplicate_check[:error]
+          return response
+        end
+
+        job = enqueue_import_job(
+          schools_data: parsed_schools[:schools],
+          user_id: current_user.id
+        )
+
+        response[:job_id] = job.job_id
+        response[:total_schools] = parsed_schools[:schools].length
+        response
+      rescue StandardError => e
+        Sentry.capture_exception(e)
+        response ||= OperationResponse.new
+        response[:error] = SchoolImportError.format_error(:unknown_error, e.message)
+        response
+      end
+
+      private
+
+      def check_duplicate_owners(schools)
+        owner_emails = schools.filter_map { |s| s[:owner_email]&.strip&.downcase }
+        duplicates = owner_emails.select { |e| owner_emails.count(e) > 1 }.uniq
+
+        if duplicates.any?
+          error = SchoolImportError.format_error(
+            :duplicate_owner_email,
+            'Duplicate owner emails found in CSV',
+            { duplicate_emails: duplicates }
+          )
+          return { error: error }
+        end
+
+        {}
+      end
+
+      def parse_csv(csv_file)
+        require 'csv'
+
+        csv_content = csv_file.read
+        return { error: SchoolImportError.format_error(:csv_invalid_format, 'CSV file is empty') } if csv_content.blank?
+
+        csv_data = CSV.parse(csv_content, headers: true, header_converters: :symbol)
+
+        if csv_data.headers.nil? || !valid_headers?(csv_data.headers)
+          return {
+            error: SchoolImportError.format_error(
+              :csv_invalid_format,
+              'Invalid CSV format. Required headers: name, website, address_line_1, municipality, country_code, owner_email'
+            )
+          }
+        end
+
+        process_csv_rows(csv_data)
+      rescue CSV::MalformedCSVError => e
+        { error: SchoolImportError.format_error(:csv_malformed, "Invalid CSV file format: #{e.message}") }
+      rescue StandardError => e
+        Sentry.capture_exception(e)
+        { error: SchoolImportError.format_error(:unknown_error, "Failed to parse CSV: #{e.message}") }
+      end
+
+      def process_csv_rows(csv_data)
+        schools = []
+        errors = []
+
+        csv_data.each_with_index do |row, index|
+          row_number = index + 2 # +2 because index starts at 0 and we skip header row
+          school_data = row.to_h
+
+          validation_errors = validate_school_data(school_data, row_number)
+          if validation_errors
+            errors << validation_errors
+            next
+          end
+
+          schools << school_data
+        end
+
+        if errors.any?
+          { error: SchoolImportError.format_row_errors(errors) }
+        else
+          { schools: schools }
+        end
+      end
+
+      def valid_headers?(headers)
+        required = %i[name website address_line_1 municipality country_code owner_email]
```

Comment:
We could define this as a constant since it's referenced multiple times, and could also use it in the `csv_invalid_format` error above